xc/common/utils/prjxray_create_place_constraints.py

```diff
@@ -200,15 +210,21 @@ def __init__(self, vpr_grid, io_locs, blif_data, roi):
 
         cmt_dict = vpr_grid.get_cmt_dict()
 
-        top_cmt_tile = next(
-            k for k, v in cmt_dict.items() if k.startswith('CLK_BUFG_TOP')
-        )
-        bot_cmt_tile = next(
-            k for k, v in cmt_dict.items() if k.startswith('CLK_BUFG_BOT')
-        )
+        try:
+            top_cmt_tile = next(
+                k for k, v in cmt_dict.items() if k.startswith('CLK_BUFG_TOP')
+            )
+            _, thresh_top_y = cmt_dict[top_cmt_tile]['canon_loc']
+        except StopIteration:
+            thresh_top_y = None
 
-        _, thresh_top_y = cmt_dict[top_cmt_tile]['canon_loc']
-        _, thresh_bot_y = cmt_dict[bot_cmt_tile]['canon_loc']
+        try:
+            bot_cmt_tile = next(
+                k for k, v in cmt_dict.items() if k.startswith('CLK_BUFG_BOT')
+            )
+            _, thresh_bot_y = cmt_dict[bot_cmt_tile]['canon_loc']
+        except StopIteration:
+            thresh_bot_y = None
 
```

Comment:
Should likely assert that either thresh_bot_y or thresh_top_y is present.  Otherwise things will likely break horribly.

Reply:
Also if graph limit is not set, both thresh_bot_y and thresh_top_y should be `not None`.

Reply:
I have added the requested changes